code/ui/.storybook/main.ts

```diff
@@ -0,0 +1,58 @@
+import vitePluginReact from '@vitejs/plugin-react';
+import type { PluginOption } from 'vite';
+import type { StorybookConfig } from '../../frameworks/react-vite/dist';
+
+const config: StorybookConfig = {
+  stories: [
+    '../manager/src/**/*.stories.@(ts|tsx|js|jsx|mdx)',
+    // '../../lib/components/src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
+    // '../../../addons/interactions/**/*.stories.@(ts|tsx|js|jsx|mdx)',
+  ],
+  addons: [
+    '@storybook/addon-links',
+    '@storybook/addon-essentials',
+    '@storybook/addon-interactions',
+  ],
+  framework: {
+    name: '@storybook/react-vite',
+    options: {},
+  },
+  viteFinal: (config) => {
+    return {
+      ...config,
+      optimizeDeps: {
+        ...config.optimizeDeps,
+        include: [
+          ...(config.optimizeDeps?.include ?? []),
+          'react-element-to-jsx-string',
+          'core-js/modules/es.regexp.flags.js',
+          'react-colorful',
+        ],
+      },
+      /*
+      This might look complex but all we're doing is removing the default set of React Vite plugins
+      and adding them back in, but with the `jsxRuntime: 'classic'` option.
+      TODO: When we've upgraded to React 18 all of this shouldn't be necessary anymore
+      */
+      plugins: [...withoutReactPlugins(config.plugins), vitePluginReact({ jsxRuntime: 'classic' })],
```

Comment:
Let's (in a follow-up PR) extract this code into the react-vite framework.
We can toggle this behavior in a frameworkOption.

@shilman perhaps this is something we should do this cycle, or the next? And maybe we need a repro for react16+vite, if we do not have one yet.

We could also say react16+vite is unsupported, but that might be a bit too aggressive? (considering we ourselves are using it 🤡